x-pack/plugins/security_solution/public/resolver/store/selectors.ts

```diff
@@ -122,3 +132,93 @@ function composeSelectors<OuterState, InnerState, ReturnValue>(
 ): (state: OuterState) => ReturnValue {
   return (state) => secondSelector(selector(state));
 }
+
+const spatiallyIndexedEntities = createSelector(
+  (state: ResolverState) => state,
+  function (state) {
+    const indexedProcessTree = indexedProcessTreeFactory(
+      dataSelectors.graphableProcesses(state.data)
+    );
+    const processNodeViewWidth = 360;
+    const processNodeViewHeight = 120;
+    const widths = dataSelectors.widthsOfProcessSubtrees(indexedProcessTree);
+
+    const positions = dataSelectors.processPositions(indexedProcessTree, widths);
+
+    const edgeLineSegments = dataSelectors.processEdgeLineSegments(
+      indexedProcessTree,
+      widths,
+      positions
+    );
+
+    const tree: rbush<IndexedEntity> = new rbush();
+    const processesToIndex: IndexedProcessNode[] = [];
+    const edgeLineSegmentsToIndex: IndexedEdgeLineSegment[] = [];
+    for (const [processEvent, position] of positions) {
+      const transformedPosition = applyMatrix3(position, dataSelectors.isometricTransformMatrix);
```

Comment:
:information_source: This is _entirely for your amusement_ and not really a serious a suggestion of any kind, but I thought you would enjoy it: https://gist.github.com/bkimmel/054eb12d9f13ce8bece83d31de20a333